### PR TITLE
Nickel/ArcadeFinal

### PR DESCRIPTION
**Pull request by NickelXu0**
<h2>Summary</h2>
 
Finished arcade game. Game name: Garden War. It's a game that two players can manipulate little witches living in the forest to compete for "territory" in their garden in limited time. 

<h2>Details</h2>

Visual style completed, add background story, soundeffect, animated characters sprite, UI, and shooting function for characters. 

<h2>Visuals</h2>

动画

<h2>How to Test</h2>

- WASD for player1 and arrow for player2. And you can shoot by pushing the button followed instruction. Your goal is to get more territory for win. Once step on territory, it will turn to yours. It can be changed once the other player step on after. 

<h2>How to Review</h2>

- If this game is interesting to play with your friends? If it's not, how should I improve my game machanic?
- I tried to make a retro style arcade game, to what extent does it achieve this goal?

### CODE REVIEW

**Comment by ekaterina392** [comment]
Hi Nickel,

I think you have created a nice multiplayer game! It was so funny to see that players can bump into each other, which I guess will add even more competition for filling the squares. Maybe you should add some time (additional 1 minute or so) as it seems too fast now, as well as make the play space bigger.

File: Assets/NickelArcade/Player.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace NickelVampireSurvive
```
Comment (jasonzli) [inline Assets/NickelArcade/Player.cs:5]: This is probably not the namespace you want 😆 `Nickel` will do just fine.

File: Assets/NickelArcade/PrintScore.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+
+public class PrintScore : MonoBehaviour
```
Comment (jasonzli) [inline Assets/NickelArcade/PrintScore.cs:6]: namespace

File: Assets/NickelArcade/SceneLoader.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class SceneLoader : MonoBehaviour
```
Comment (jasonzli) [inline Assets/NickelArcade/SceneLoader.cs:6]: namespace

File: Assets/NickelArcade/TerritoryChangeColor.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TerritoryChangeColor : MonoBehaviour
```
Comment (jasonzli) [inline Assets/NickelArcade/TerritoryChangeColor.cs:5]: namespace

File: Assets/NickelArcade/Timer.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+
+public class Timer : MonoBehaviour
```
Comment (jasonzli) [inline Assets/NickelArcade/Timer.cs:6]: name space

File: Assets/NickelArcade/Timer.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            else
+            {
+                Debug.Log("time is up");
+                TimeLeft = 0;
+                TimerOn = false;
+            }
+        }
+    }
+
+    void updateTimer(float currentTime)
```
Comment (jasonzli) [inline Assets/NickelArcade/Timer.cs:37]: capital u `UpdateTimer`

File: Assets/NickelArcade/PrintScore.cs [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    void Start()
+    {
+        printPlayer1Score.GetComponent<TextMeshPro>().text = "";
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+        printPlayer1Score.GetComponent<TMP_Text>().text = "Red--" +TerritoryChangeColor.player1Score.ToString();
```
Comment (jasonzli) [inline Assets/NickelArcade/PrintScore.cs:20]: Does this really need to run every `Update`?

File: ProjectSettings/TagManager.asset [hunk 8]
```diff
@@ -5,6 +5,8 @@ TagManager:
   serializedVersion: 2
   tags:
   - Enemy
+  - Player2
```
Comment (jasonzli) [inline ProjectSettings/TagManager.asset:8]: Be cautious about the potential for conflicts here

File: ProjectSettings/InputManager.asset [hunk 9]
```diff
@@ -21,6 +21,38 @@ InputManager:
     type: 0
     axis: 0
     joyNum: 0
+  - serializedVersion: 3
```
Comment (jasonzli) [inline ProjectSettings/InputManager.asset:24]: You will likely need to redo these after a merge conflict resolution

File: ProjectSettings/EditorBuildSettings.asset [hunk 10]
```diff
@@ -18,18 +18,24 @@ EditorBuildSettings:
...
     guid: 47ac0cb2a332846768133bda92b1cbb0
   - enabled: 0
-    path: Assets/AishaBikebayeva/Scenes/WerewolfReviversWin.unity
+    path: 
     guid: e46eadf6808744836bf2147d7a4af55c
   - enabled: 1
     path: Assets/JasonLi/ARCADE/BasicScene.unity
     guid: a9f602ea32ef9ce469489c9b7cf9da64
+  - enabled: 1
+    path: Assets/NickelArcade/StartPage.unity
```
Comment (jasonzli) [inline ProjectSettings/EditorBuildSettings.asset:36]: These will likely run into conflicts

**Review by jasonzli — CHANGES_REQUESTED** [review]
Overall a good setup, but I recommend putting your scripts into a sub folder as well as your visual assets/aniamtions into another. Then name your scene something more descriptive and specific to you.

**Comment by jasonzli** [comment]
Your PR description is also needs to be completed

**Review by zaksaja — APPROVED** [review]

**Comment by NickelXu0** [comment]
Thank you for your advice! I will definitely try it!

ekaterina392 ***@***.***>于2022年9月12日 周一下午6:04写道：

**Pull request merged**